src/main/java/io/jenkins/plugins/folderauth/FolderAuthorizationStrategyManagementLink.java

```diff
@@ -443,4 +446,131 @@ public void doRemoveSidFromAgentRole(@QueryParameter(required = true) String rol
         FolderAuthorizationStrategyAPI.removeSidFromAgentRole(sid, roleName);
         redirect();
     }
+
+    /**
+     * API Method to get an {@link AgentRole}
+     * Example: {@code curl -X GET 'http://localhost:8080/jenkins/folder-auth/getAgentRole?name=agentSmithRole}
+     *
+     * @param name name of the role (single, no list)
+     */
+    @Restricted(NoExternalUse.class)
+    public void doGetAgentRole(@QueryParameter(required = true) String name) throws IOException {
+        Jenkins.get().checkPermission(Jenkins.ADMINISTER);
+        JSONObject responseJson = new JSONObject();
+        AgentRole role = FolderAuthorizationStrategyAPI.getAgentRole(name);
+        if (role != null) {
+            responseJson.put("name", role.getName());
```

Comment:
Would it be possible to let json-lib handle converting the AbstractRole objects to JSONObjects using [`JSONObject#fromObject`](http://json-lib.sourceforge.net/apidocs/jdk15/net/sf/json/JSONObject.html#fromObject(java.lang.Object))?

Reply:
To use `JSONObject#fromObject`, the role class must be of type string (json formatted), map, etc. The current class doesn't so it'll require a refactor for the class to extend either of these classes it to work. I'll keep it as is? Or we can